### PR TITLE
Travis/docker build fixes

### PR DESCRIPTION
**Pull request by cskiraly**
Fixes to the build process. Still not uniform with other repos, but getting close.

### CODE REVIEW

**Comment by cskiraly** [comment]
@craig-mulligan anything against merging this?

**Comment by craigmulligan** [comment]
Hey @cskiraly sorry, missed this. I'll look over it this afternoon, sorry for the delays.

File: .travis.yml [hunk 1]
```diff
@@ -57,6 +59,10 @@ after_success:
       docker login -u="$DOCKER_USERNAME" -p="$DOCKER_PASSWORD";
       docker push $DOCKER_IMAGE:$DOCKER_TAG;
     fi
+  - if [ "$DOCKER_TAG" == "master" ]; then
```
Comment (craigmulligan) [inline .travis.yml:62]: Why is this necessary? If I recall it pushes on any branch that has a tag? And master should only get pushed when it is an actual release AKA has a git tag.

File: Dockerfile [hunk 2]
```diff
@@ -1,10 +1,6 @@
-FROM resin/raspberrypi3-node:7.6-20170318
+FROM resin/raspberry-pi3-node:7.8.0-20170426
 
-RUN wget \
-	"https://caddyserver.com/download/build?os=linux&arch=arm" \
-	-O /tmp/caddy.tar.gz \
-	&& tar -xzf /tmp/caddy.tar.gz -C /usr/local/bin/ \
-	&& rm /tmp/caddy.tar.gz
+RUN wget -qO- https://getcaddy.com | bash
```
Comment (craigmulligan) [inline Dockerfile:3]: Does this get the correct arch package based on the request?

File: .travis.yml (hunk 1 above)
Comment (cskiraly) [reply .travis.yml:62]: The image structure we are aiming for:
- tagged releases, with dockertag = git tag
- head of each branch (i.e. moving target), with dockertag = git branch name
- PRs, with dockertag = pr-<targetbranch>-<id>
- dockertag latest = dockertag master.
This part of the code is for the last point.

File: Dockerfile (hunk 2 above)
Comment (cskiraly) [reply Dockerfile:3]: yes it does. Chech code at https://getcaddy.com/

File: .travis.yml (hunk 1 above)
Comment (craigmulligan) [reply .travis.yml:62]: I agree. (I like the pr prefix name'ing)

What I'm saying is that every commit on master would be tagged (so there is no need for keeping it up to date) You are just pushing duplicate images with different tags.

Comment (cskiraly) [reply .travis.yml:62]: indeed, it is just a duplicate tag. But since "latest" is a special tag in docker terminology, omitting it would cost us. The best option I've found was to make it equivalent to master. Other ideas?

File: Dockerfile (hunk 2 above)
Comment (craigmulligan) [reply Dockerfile:3]: Nice!

File: .travis.yml (hunk 1 above)
Comment (craigmulligan) [reply .travis.yml:62]: Oh right I forgot about latest, I think we should just stop using it because it often isn't what is says it is. But in the meantime let's do the double push.

**Review by craigmulligan — APPROVED** [review]
lgtm, can you squash the commits before merging?
